components/camera/videosource/webcam.go

```diff
@@ -45,7 +46,19 @@ func init() {
 			if !ok {
 				return nil, utils.NewUnexpectedTypeError(attrs, config.ConvertedAttributes)
 			}
-			return NewWebcamSource(ctx, config.Name, attrs, logger)
+			cameraSource, err := NewWebcamSource(ctx, config.Name, attrs, logger)
+			if err != nil {
+				// If we are on a Jetson Orin AGX, we need to validate driver and daughterboard setup
+				osInfo, osErr := jetsoncamera.DetectOSInformation()
```

Comment:
From @penguinland 

The rest of this file appears to be hardware-agnostic. If this code runs on a non-Jetson board, what will/should happen?

Reply:
Tested on mac and pi.

mac: `DetectOSInformation` will return `osError`, so we simply return err from `NewWebcamSource` and do not check for daughterboard/driver.

pi: returns successfully `osInfo`, check for device name which is not `OrinAGX`, so we again simply return error from `NewWebcamSource`